contracts/core/Dispatcher.sol

```diff
@@ -8,6 +8,18 @@ import '../interfaces/IbcDispatcher.sol';
 import {IbcChannelReceiver, IbcPacketReceiver} from '../interfaces/IbcReceiver.sol';
 import '../interfaces/ConsensusStateManager.sol';
 
+struct PackedNextSequence {
+    uint64 send;
```

Comment:
This struct packing is brilliant, thanks for this @reednaa !

Reply:
Note that this doesn't actually save as much gas as you might imagine. The slots are used independently of each other, so you don't save on warming up the slots which often contribute the biggest amount.
However, you do still save a lot on only changing a 0 slot to non-zero 1 time instead of 3 times. That is (if I recall correctly) roughly 2x~20k gas saves.